ob1k-cache/src/main/java/com/outbrain/ob1k/cache/memcache/folsom/MemcachedClient.java

```diff
@@ -35,17 +37,29 @@
  */
 public class MemcachedClient<K, V> implements TypedCache<K, V> {
 
+  private static final Logger log = LoggerFactory.getLogger(MemcachedClient.class);
+
   private final MemcacheClient<V> folsomClient;
   private final CacheKeyTranslator<K> keyTranslator;
   private final int expirationSeconds;
   // TODO should we create a dedicated executor?
   private final Executor executor = ComposableFutures.getExecutor();
+  private final String cacheName;
 
   // TODO add docs (especially about the expiration rules)
   public MemcachedClient(final MemcacheClient<V> folsomClient, final CacheKeyTranslator<K> keyTranslator, final long expiration, final TimeUnit timeUnit) {
+    this(folsomClient, keyTranslator, expiration, timeUnit, "UNKNOWN");
+  }
+
+  public MemcachedClient(final MemcacheClient<V> folsomClient,
+                         final CacheKeyTranslator<K> keyTranslator,
+                         final long expiration,
+                         final TimeUnit timeUnit,
+                         final String cacheName) {
```

Comment:
I really dislike this, but I understand why we're reluctant to fix the references :P